storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -3557,6 +3570,14 @@ public ByteBuffer downloadBlobChunk(String session) throws AuthorizationExceptio
     @Override
     public void deleteBlob(String key) throws AuthorizationException, KeyNotFoundException, TException {
         try {
+            String topoName = ConfigUtils.getIdFromBlobKey(key);
+            if (topoName != null) {
+                if (isTopologyActiveOrActivating(stormClusterState, topoName)) {
+                    String message = "Attempting to delete blob " + key + " from under active topology " + topoName;
+                    LOG.warn(message);
+                    throw new TException(message);
```

Comment:
Could we make this an AuthorizationException?

Reply:
I thought of that and wondered if that would be confusing to the users since it is a temporary exception.  Willing to change it though.

Reply:
Yeah may be better to have a specific exception to clearly state the situation.

Reply:
Thinking about it I think adding in a new exception would be the best way to go.  All of the APIs throw a TException so adding in a new more specific TException would not break any clients any worse than throwing a raw TException would.

Reply:
Makes sense to me.  I prefer this to an AuthorizationException.